packages/create-llama/questions.ts

```diff
@@ -373,108 +405,155 @@ export const askQuestions = async (
+          process.exit(1);
+        } else {
+          program.dataSource = {
+            type: fs.lstatSync(filePath).isDirectory() ? "folder" : "file",
+            config: {
+              path: filePath,
+            },
+          };
         }
-      }
-    }
-
-    if (program.dataSource?.type === "web" && program.framework === "fastapi") {
-      let { baseUrl } = await prompts(
-        {
-          type: "text",
-          name: "baseUrl",
-          message: "Please provide base URL of the website:",
-          initial: "https://www.llamaindex.ai",
-        },
-        handlers,
-      );
-      try {
-        if (!baseUrl.includes("://")) {
-          baseUrl = `https://${baseUrl}`;
+      } else {
+        let choices = [
+          {
+            title: "No data, just a simple chat",
+            value: "simple",
+          },
+          { title: "Use an example PDF", value: "exampleFile" },
+        ];
+        if (process.platform === "win32" || process.platform === "darwin") {
+          if (process.platform === "win32" || process.platform === "darwin") {
+            choices.push({
+              title: `Use a local file (${supportedContextFileTypes})`,
+              value: "localFile",
+            });
+            choices.push({
+              title: `Use a local folder`,
+              value: "localFolder",
+            });
+          }
         }
-        let checkUrl = new URL(baseUrl);
-        if (checkUrl.protocol !== "https:" && checkUrl.protocol !== "http:") {
-          throw new Error("Invalid protocol");
+        if (program.framework === "fastapi") {
+          choices.push({ title: "Use website content", value: "web" });
         }
-      } catch (error) {
-        console.log(
-          red(
-            "Invalid URL provided! Please provide a valid URL (e.g. https://www.llamaindex.ai)",
-          ),
+
+        const { dataSource } = await prompts(
+          {
+            type: "select",
+            name: "dataSource",
+            message: "Which data source would you like to use?",
+            choices: choices,
+            initial: 1,
+          },
+          handlers,
         );
-        process.exit(1);
+        // Initialize with default config
+        program.dataSource = getPrefOrDefault("dataSource");
+        if (program.dataSource) {
+          switch (dataSource) {
+            case "simple":
+              program.engine = "simple";
+              break;
+            case "exampleFile":
+              program.engine = "context";
+              // example file is a context app with dataSource.type = file but has no config
+              program.dataSource = { type: "file", config: {} };
+              break;
+            case "localFile":
+              program.engine = "context";
+              program.dataSource.type = "file";
+              program.dataSource.config = {
+                path: await selectLocalContextData(program.dataSource),
```

Comment:
i guess only the type is needed?
```suggestion
                path: await selectLocalContextData(program.dataSource.type),
```